.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,71 @@
+name: Benchmark
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: 'Branch to benchmark'
+        required: true
+        default: 'main'
+        type: string
+  schedule:
+    - cron: '0 0 * * SAT'
+
+permissions:
+  contents: read
+
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      # with:
+      #   ref: ${{ github.event_name == 'workflow_dispatch' && inputs.branch || 'main' }}
```

Comment:
nit: can be removed

Reply:
Actually needs to be uncommented. I had it commented so I could test it